### PR TITLE
VOLDEV-145 - Implement article header changes and new rail module

### PR DESCRIPTION
**Pull request by onei**
**Awaiting review by VolDevs**

Ticket: https://wikia-inc.atlassian.net/browse/VOLDEV-145

**Changes:**
- Move edit buttons (and other buttons) back above the page header border. This also fixes CONCF-501.
- Move the page count tally to it's own sidebar module, and move the contribute button to said module as well. Special:Videos keeps it's own tally (and no longer has the page share buttons).

**Cleanup includes:**
- Fixed various code issues, e.g. `array()` -> `[]`, `wfMsg*` -> `wfMessage`, whitespace, etc. to conform with the style guides.
- Altered the `PageHeaderIndexAfterActionButtonPrepared` and `PageHeaderIndexExtraButtons` hooks to only modify the intended variables to reduce the potential for subtle, undocumented behaviour/bugs.
- Moved extension specific buttons to hook callbacks in `PageHeader_Index.php` (per a TODO comment), thus adding `SpecialVideosHooks.php` and `WikiaNewFilesHooks.php`.
- Added icons to the user talk archive edit button (seen when Wall is enabled).
- Fixed inconsistencies in `DefaultSettings.php` (simply picked what seemed to be the more predominant method).
- Removed unused sass, e.g. removing styling related to search in the page header.
- Removed use of globals in controller methods in favour of accessing them via `F::app()` (or `$this` if non-static.
- Overhauled logic for `PageHeaderContoller` and it's templates.
- Prevented modification of URLs for the share buttons in the page header for hub wikis (always use database defaults).
- Deleted `PageHeader_EditBox.php` as it was unused (and it's associated method).

**To be fixed before merging:**
- Remove temp code in `PageHeaderController::executeHubs` (sass loading) and `ContributeController::index` (commented out code for testing wikis with high page counts)
- Move the new sidebar module to the correct place in `BodyController::getRailModuleList` (awaiting confirmation of how it's meant to behave in ticket).
- Rebase to fix conflicts (largely related to curatedcontent).

### CODE REVIEW

File: extensions/wikia/WikiaNewFiles/WikiaNewFiles_setup.php [hunk 1]
```diff
@@ -2,29 +2,29 @@
...
 	'url'            => 'https://github.com/Wikia/app/tree/dev/extensions/wikia/WikiaNewFiles',
-);
-
-$dir = dirname( __FILE__ );
+];
 
-// Translations
-$wgExtensionMessagesFiles["WikiaNewFiles"] =  "{$dir}/SpecialNewFiles.i18n.php";
+// i18n
+$wgExtensionMessagesFiles["WikiaNewFiles"] =  __DIR__ . '/SpecialNewFiles.i18n.php';
```
Comment (Warkot) [inline extensions/wikia/WikiaNewFiles/WikiaNewFiles_setup.php]: Double quotes spoted. :telescope:

File: skins/oasis/modules/PageHeaderController.class.php [hunk 2]
```diff
@@ -460,150 +433,106 @@ public function executeEditPage() {
...
+			// show only "My Tools" dropdown on toolbar
+			$wg->SuppressToolbar = true;
 		}
 
 		// choose header message
 		if ( $isPreview ) {
 			$titleMsg = 'oasis-page-header-preview';
-		}
-		else if ( $isShowChanges ) {
+		} else if ( $isShowChanges ) {
```
Comment (Warkot) [inline skins/oasis/modules/PageHeaderController.class.php]: If you're cleaning up, shouldn't these be converted from `else if` to `elseif`? http://php.net/manual/pl/control-structures.elseif.php

Comment (onei) [inline skins/oasis/modules/PageHeaderController.class.php]: Thanks, I wasn't aware of that one :)

File: extensions/wikia/Forum/ForumHooksHelper.class.php [hunk 3]
```diff
@@ -237,20 +237,30 @@ static function onGetUserPermissionsErrors(Title &$title, User &$user, $action,
...
+		// check namespace(s)
 		if ( $ns == NS_FORUM || $ns == NS_FORUM_TALK ) {
 			if ( !static::canEditOldForum( $app->wg->User ) ) {
-				$action = array( 'class' => '', 'text' => wfMessage( 'viewsource' )->escaped(), 'href' => $title->getLocalUrl( array( 'action' => 'edit' ) ), 'id' => 'ca-viewsource', 'primary' => 1 );
-				$response->setVal( 'actionImage', MenuButtonController::LOCK_ICON );
-				$response->setVal( 'action', $action );
+				$action = [
+					'class' => '',
+					'text' => wfMessage( 'viewsource' )->escaped(),
+					'href' => $title->getLocalUrl( ['action' => 'edit'] ),
```
Comment (Warkot) [inline extensions/wikia/Forum/ForumHooksHelper.class.php]: When defining an array, put spaces inside the brackets. Don't put spaces there when pointing to an index. Therefore it should be `[ 'action' => 'edit' ]`, but `$button['action']` below is OK. Wikia Guidelines for reference: https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#array-syntax.

File: skins/oasis/modules/PageHeaderController.class.php (hunk 2 above)
Comment (Warkot) [inline skins/oasis/modules/PageHeaderController.class.php]: _¡De nada!_ :smiley: Some more here: http://stackoverflow.com/questions/3662412/are-elseif-and-else-if-completely-synonymous

**Pull request closed without merging**